File: src/main/java/com/musinsa/subject/model/price/LowestTotalPriceBrandWithCategoryDetailsResponse.java
package com.musinsa.subject.model.price;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
@Builder
@AllArgsConstructor
public class LowestTotalPriceBrandWithCategoryDetailsResponse {

    public record CategoryDetail(String categoryName, String productPrice) {
    }

    @Getter
    @Setter
    @Builder
    public static class LowestPrice {

        private String brandName;
        private List<CategoryDetail> categories;
        private String totalPrice;

    }

    public LowestPrice lowestPrice;

}

File: src/main/kotlin/com/musinsa/subject/model/price/LowestTotalPriceBrandWithCategoryDetailsResponse.kt
package com.musinsa.subject.model.price

class LowestTotalPriceBrandWithCategoryDetailsResponse(
    var lowestPrice: LowestPrice? = null
) {
    data class CategoryDetail(
        val categoryName: String,
        val productPrice: String
    )

    class LowestPrice(
        var brandName: String? = null,
        var categories: List<CategoryDetail>? = null,
        var totalPrice: String? = null
    )
}
